core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala

```diff
@@ -114,175 +170,183 @@ trait RawlsBillingProjectComponent {
-    }
+        .withProjectName(billingProjectName)
+        .setBillingProjectSpendConfiguration(datasetName, tableName, datasetGoogleProject)
 
-    def clearBillingProjectSpendConfiguration(billingProjectName: RawlsBillingProjectName): WriteAction[Int] = {
+    def clearBillingProjectSpendConfiguration(billingProjectName: RawlsBillingProjectName): WriteAction[Int] =
       setBillingProjectSpendConfiguration(billingProjectName, None, None, None)
-    }
 
-    def getBillingProjectSpendConfiguration(billingProjectName: RawlsBillingProjectName): ReadAction[Option[BillingProjectSpendExport]] = {
-      uniqueResult[RawlsBillingProjectRecord](findBillingProjectByName(billingProjectName)).map(_.map(unmarshalBillingProjectSpendExport))
-    }
+    def getBillingProjectSpendConfiguration(billingProjectName: RawlsBillingProjectName): ReadAction[Option[BillingProjectSpendExport]] =
+      rawlsBillingProjectQuery
+        .withProjectName(billingProjectName)
+        .result
+        .map(_.headOption.map(RawlsBillingProjectRecord.toBillingProjectSpendExport))
 
-    def insertOperations(operations: Seq[RawlsBillingProjectOperationRecord]): WriteAction[Unit] = {
+    def insertOperations(operations: Seq[RawlsBillingProjectOperationRecord]): WriteAction[Unit] =
       (rawlsBillingProjectOperationQuery ++= operations).map(_ => ())
-    }
 
-    def updateOperations(operations: Seq[RawlsBillingProjectOperationRecord]): WriteAction[Seq[Int]] = {
+    def updateOperations(operations: Seq[RawlsBillingProjectOperationRecord]): WriteAction[Seq[Int]] =
       DBIO.sequence(operations.map(rec => rawlsBillingProjectOperationQuery.filter(x => x.projectName === rec.projectName && x.operationName === rec.operationName).update(rec)))
-    }
 
-    def loadOperationsForProjects(projectNames: Seq[RawlsBillingProjectName], operationName: GoogleOperationName): ReadAction[Seq[RawlsBillingProjectOperationRecord]] = {
+    def loadOperationsForProjects(projectNames: Seq[RawlsBillingProjectName], operationName: GoogleOperationName): ReadAction[Seq[RawlsBillingProjectOperationRecord]] =
       rawlsBillingProjectOperationQuery.filter(x => x.projectName.inSetBind(projectNames.map(_.value)) && x.operationName === operationName).result
-    }
+  }
 
-    private def marshalBillingProject(billingProject: RawlsBillingProject): RawlsBillingProjectRecord = {
-      RawlsBillingProjectRecord(billingProject.projectName.value, billingProject.status.toString, billingProject.billingAccount.map(_.value), billingProject.message, billingProject.cromwellBackend.map(_.value), billingProject.servicePerimeter.map(_.value), billingProject.googleProjectNumber.map(_.value), billingProject.invalidBillingAccount, billingProject.spendReportDataset.map(_.value), billingProject.spendReportTable.map(_.value), billingProject.spendReportDatasetGoogleProject.map(_.value))
-    }
+  implicit class RawlsBillingProjectExtensions(query: RawlsBillingProjectQuery) {
```

Comment:
Cool idea with these extension methods

Reply:
Thanks but I don't think I can't claim the credit - I think this is how slick implements many of the operations on `Query`!